### ISSUE TITLE
order_by nested looks buggy

### ISSUE THREAD

**Issue by sm0k** [tortoise/tortoise-orm#326]
**Describe the bug**
Order by on nested field seems buggy

**To Reproduce**
```
from tortoise.models import Model
from tortoise import fields

class Tournament(Model):
    id = fields.IntField(pk=True)
    name = fields.TextField()

    def __str__(self):
        return self.name


class Event(Model):
    id = fields.IntField(pk=True)
    name = fields.TextField()
    tournament = fields.ForeignKeyField('models.Tournament', related_name='events')
    participants = fields.ManyToManyField('models.Team', related_name='events', through='event_team')

    def __str__(self):
        return self.name


class Team(Model):
    id = fields.IntField(pk=True)
    teamname = fields.TextField()

    def __str__(self):
        return self.teamname

from tortoise import Tortoise

async def init():
    # Here we connect to a SQLite DB file.
    # also specify the app name of "models"
    # which contain models from "app.models"
    await Tortoise.init(db_url="sqlite://:memory:", modules={"models": ["__main__"]})
    # Generate the schema
    await Tortoise.generate_schemas()

    # Create instance by save
    tournament = Tournament(name='New Tournament')
    await tournament.save()

    # Or by .create()
    await Event.create(name='Without participants', tournament=tournament)
    event = await Event.create(name='Test', tournament=tournament)
    participants = []
    for i in range(2):
        team = await Team.create(teamname='Team {}'.format(i + 1))
        participants.append(team)

    # M2M Relationship management is quite straightforward
    # (also look for methods .remove(...) and .clear())
    await event.participants.add(*participants)

    # You can query related entity just with async for
    async for team in event.participants:
        pass

    # After making related query you can iterate with regular for,
    # which can be extremely convenient for using with other packages,
    # for example some kind of serializers with nested support
    for team in event.participants:
        pass

    # Or you can make preemptive call to fetch related objects
    selected_events = await Event.filter(
        participants=participants[0].id
    ).prefetch_related('participants', 'tournament')

    # Tortoise supports variable depth of prefetching related entities
    # This will fetch all events for team and in those events tournaments will be prefetched
    await Team.all().prefetch_related('events__tournament')

    # You can filter and order by related models too
    await Tournament.filter(
        events__name__in=['Test', 'Prod']
    ).order_by('-events__participants__teamname').distinct()
import asyncio
if __name__ == "__main__":
    asyncio.run(init())

```
This code is the basic sample, the only thing modified is change  Team.name to team.teamname

This line crashes:
    await Tournament.filter(
        events__name__in=['Test', 'Prod']
    ).order_by('-events__participants__teamname').distinct()

with the following exception:
`tortoise.exceptions.FieldError: Unknown field teamname for model Tournament.`

I cant find a way to order by nested field correctly.

In fact, the basic samples seems to work ONLY because Tournament.name exists.
 
**Expected behavior**
Should order results by Team.teamname

**Comment by sm0k** [comment]
https://github.com/tortoise/tortoise-orm/blob/a3f0ff621a449362f11461d670665afc2f20d31a/tortoise/queryset.py#L187

`field_object = self.model._meta.fields_map.get(field_name)`

In case of a nested call, should'nt 'model' be used here, instead of 'self.model'?

**Comment by sm0k** [comment]
Here is a quick and dirty patch:
```
                if self.model != model:
                    field_object = model._meta.fields_map.get(field_name)
                else:
                    field_object = self.model._meta.fields_map.get(field_name)
```
I can do it clean and open a PR if you want?

**Comment by grigi** [comment]
Hi, if you want to do a PR, you are welcome :-)
Preferably with at least one test to prevent future regressions.

**Comment by grigi** [comment]
Released v0.16.3 & v0.15.21 with this fix.
